src/internal/ans.ts

```diff
@@ -171,5 +171,207 @@ export async function registerName(args: RegisterNameParameters): Promise<InputS
     return transaction as InputSingleSignerTransaction;
   }
 
-  throw new Error(`Policy ${expiration.policy} is not supported yet`);
+  // We are a subdomain
+  if (!subdomainName) {
+    throw new Error(`${expiration.policy} requires a subdomain to be provided.`);
+  }
+
+  let tldExpiration = await getExpiration({ aptosConfig, name: domainName });
+  if (!tldExpiration) {
+    throw new Error("The domain does not exist");
+  }
+  // The contract gives us seconds, but JS expects milliseconds
+  tldExpiration *= 1000;
+
+  const expirationDateInMillisecondsSinceEpoch =
+    expiration.policy === "subdomain:independent" ? expiration.expirationDate.valueOf() : tldExpiration;
+
+  if (expirationDateInMillisecondsSinceEpoch > tldExpiration) {
+    throw new Error("The subdomain expiration time cannot be greater than the domain expiration time");
+  }
+
+  const transaction = await generateTransaction({
+    aptosConfig,
+    sender: sender.accountAddress.toString(),
+    data: {
+      function: `${routerAddress}::router::register_subdomain`,
+      functionArguments: [
+        new MoveString(domainName),
```

Comment:
you should be able to use primitive values and not to instantiate with those classes

Reply:
have you looked at it ^ ?